### PR TITLE
Fix register return type

### PR DESCRIPTION
**Pull request by stramel**
`ref` typings didn't match what we were returning.

Going to test this change locally post-build.

### CODE REVIEW

File: src/useForm.ts [hunk 1]
```diff
@@ -404,8 +405,8 @@ export default function useForm<
     );
   }
 
-  function __registerIntoFieldsRef(
-    elementRef: Ref,
+  function __registerIntoFieldsRef<Element extends ElementLike>(
```
Comment (bluebill1049) [inline src/useForm.ts]: didn't know this `ElementLike`,pretty cool

Comment (stramel) [reply src/useForm.ts]: I made it 😝 to better match that we allow `{ name: string}` instead of full element objects.

**Comment by bluebill1049** [comment]
Please do testing on this :)

**Comment by stramel** [comment]
Yeah definitely needs more work still.

**Pull request merged**